frontend/compositions/map/useData.tsx

```diff
@@ -0,0 +1,46 @@
+import { Feature } from "geojson"
+import { useMemo, useState } from "react"
+import uscities from "./uscities.json"
+
+export interface GeoJson {
+  type: string
+  features: Feature[]
+}
+
+export interface Filter {
+  property: string
+  value: number | string
+}
+
+export interface Data {
+  features: Feature[]
+  filter: Filter
+  setFilterProperties: (filter: Filter) => void
+}
+
+export default function useData(): Data {
+  // using temp dummy data set
+  const data: GeoJson = uscities as GeoJson
+
+  const [filterProperties, setFilterProperties] = useState<Filter>({
+    property: "state_name",
+    value: "",
+  })
+
+  return useMemo(() => {
+    const { property, value } = filterProperties
+
+    const filteredFeatures =
+      value === ""
+        ? ([] as Feature[])
+        : value === "all"
+        ? data.features
+        : data.features.filter((d: Feature) => d.properties[property] === value)
+
+    return {
+      features: filteredFeatures,
+      filter: filterProperties,
+      setFilterProperties,
```

Comment:
sorry for another nitpick! Trailing comma

Reply:
I thought trailing commas were in this year? xD